### DISCUSSION TITLE
Examples of refinement criteria on curl of B

### DISCUSSION THREAD

**Discussion by kurtsansom** [PrincetonUniversity/athena#426] [Q&A]
Can someone point me to where I could grab the curl of the magnetic field? I would like to use it as a refinement criteria.

Is it stored somewhere or would you have to calculate it?

**Comment by c-white** [comment]
curl(B) is only directly used when magnetic diffusivity is turned on (stored in the private variable `jedge_` in the `FieldDiffusion` object); otherwise it is never calculated.

I suggest simply calculating the curl even if diffusivity is being used, as derived quantities are volatile and might not be in the state or updated to the time you think they should be. When the refinement function is called, the primitive and conserved hydro variables and the face-centered fields are guaranteed to be up to date, including ghost zones.

There are currently no example problem generators that calculate the curl, but many that calculate gradients or second derivatives.

It's also worth considering whether refinement should trigger on curl(B) or curl(B) \* delta\_x. That is, how do you want the variable you are inspecting to scale with resolution?
